forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -175,6 +178,187 @@ class DSRG_MRPT2 : public MASTER_DSRG {
+    // Some variables may be redundant thus need further elimination
+    /// Set ambit tensor labels
+    void set_ambit_space();
+    /// Set tensors
+    void set_tensor();
+    /// Set density
+    void set_density();
+    /// Set Fock matrix
+    void set_fock();
+    /// Set Hamiltonian
+    void set_h();
+    /// Set two-electron integrals
+    void set_v();
+    /// Set CI-relevant integrals
+    void set_ci_ints();
+
+    /// Set MO space environment and global variables
+    void set_global_variables();
+    size_t nmo;
+    size_t ncore;
+    size_t nvirt;
+    size_t na;
+    size_t nirrep;
+
+    const bool PT2_TERM = true;
+    const bool X1_TERM = true;
+    const bool X2_TERM = true;
+    const bool X3_TERM = true;
+    const bool X4_TERM = true;
+    const bool X5_TERM = true;
+    const bool X6_TERM = true;
+    const bool X7_TERM = true;
+    const bool CORRELATION_TERM = true;
+
+    void set_dsrg_tensor();
+
+    /// Write the Lagrangian
+    void write_lagrangian();
+    /// Write spin_dependent one-RDMs coefficients
+    void write_1rdm_spin_dependent();
+    /// Write spin_dependent two-RDMs coefficients using IWL
+    void write_2rdm_spin_dependent();
+    /// TPDM backtransform
+    void tpdm_backtransform();
+
+    void set_multiplier();
+
+    void solve_z();
+    void set_sigma();
+    void set_xi();
+    void set_tau();
+    void set_kappa();
+    void set_z();
+    void set_alpha();
+    void set_CI();
+    void set_b();
+    void set_z_cc();
+    void set_z_vv();
+    void set_z_aa_diag();
+    void set_w();
+
+    size_t ndets;
+
+    /// List of core MOs (Correlated)
+    std::vector<size_t> core_mos;
+    /// List of active MOs (Correlated)
+    std::vector<size_t> actv_mos;
+    /// List of virtual MOs (Correlated)
+    std::vector<size_t> virt_mos;
+    /// List of core MOs (Absolute)
+
+    std::vector<size_t> core_all;
+    /// List of active MOs (Absolute)
+    std::vector<size_t> actv_all;
+    std::vector<size_t> virt_all;
+
+    /// List of relative core MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        core_mos_relative;
+    /// List of relative active MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        actv_mos_relative;
+    /// List of relative virtual MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        virt_mos_relative;
+
+    /// Dimension of different irreps
+    psi::Dimension irrep_vec;
+
+    /// Two-body denisty tensor
+    ambit::BlockedTensor Gamma2_;
+    /// Lagrangian tensor
+    ambit::BlockedTensor W;
+
+    // core Hamiltonian
+    ambit::BlockedTensor H;
+    // two-electron integrals
+    ambit::BlockedTensor V;
+    // Fock matrix
+    ambit::BlockedTensor F;
+
+    ambit::BlockedTensor Eeps1;
```

Comment:
Every single intermediate needs to be defined (mathematically) in the header file. If I'm reading your code and trying to figure out what `Eeps1` is, the header is the place I'd look.

Reply:
added in .h files